manager/controlapi/secret.go

```diff
@@ -157,12 +169,16 @@ func (s *Server) ListSecrets(ctx context.Context, request *api.ListSecretsReques
 //   or if the secret data is too long or contains invalid characters.
 // - Returns an error if the creation fails.
 func (s *Server) CreateSecret(ctx context.Context, request *api.CreateSecretRequest) (*api.CreateSecretResponse, error) {
+	err := s.populateSecretFromPlugin(ctx, request.Spec)
```

Comment:
@cpuguy83 I've abstracted the plugin/driver initialization and setup with a new `drivers` package. This package can be used to load any type of plugin in Swarmkit. Let me know what you think.